### PR TITLE
Update recommended Dockerfile

### PR DESCRIPTION
**Pull request by vshlapakov**
The PR fixes discovering logic for custom python scripts.

Review, please.

### CODE REVIEW

File: shub_image/init.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "WORKDIR %s" % DOCKER_APP_DIR,
     "$requirements",
     "COPY . %s" % DOCKER_APP_DIR,
+    "RUN cd %s && python setup.py install" % DOCKER_APP_DIR,
```
Comment (chekunkov) [inline shub_image/init.py]: > The WORKDIR instruction sets the working directory for any RUN, CMD, ENTRYPOINT, COPY and ADD instructions that follow it in the Dockerfile

why you need cd?

File: shub_image/init.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
...
 ]
```
Comment (chekunkov) [inline shub_image/init.py]: wouldn't it be cleaner to have a multiline string template here, something like

```python
TEMPLATE = """
FROM $base_image
$system_deps
$system_env
RUN mkdir -p {docker_app_dir}
WORKDIR {docker_app_dir}
...
""".format(docker_app_dir=DOCKER_APP_DIR)
```

File: shub_image/init.py [hunk 3]
```diff
@@ -140,7 +141,6 @@ def _wrap(text):
 def _format_system_env(settings_module):
     rows = ['ENV TERM xterm']
     if settings_module:
-        rows.append('ENV PYTHONPATH $PYTHONPATH:%s' % DOCKER_APP_DIR)
```
Comment (chekunkov) [inline shub_image/init.py]: PYTHONPATH is removed because of `setup.py install` or any other reason?

File: shub_image/init.py (hunk 2 above)
Comment (vshlapakov) [inline shub_image/init.py]: Good idea, thanks

File: shub_image/init.py (hunk 3 above)
Comment (vshlapakov) [inline shub_image/init.py]: Right, it affects discovering a distribution and its scripts:
with `/app` in `PYTHONPATH` and package installed with `setup.py install`- there're 2 versions of the same package in sys path, and only last one has scripts visible for `distribution.run_script` [method](https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/blob/master/sh_scrapy/crawl.py#L112).

File: shub_image/init.py [hunk 4]
```diff
@@ -93,8 +95,7 @@ def cli(project, base_image, base_deps, add_deps, requirements):
         'requirements': _format_requirements(project_dir, requirements),
     }
     values = {key: value if value else '' for key, value in values.items()}
-    template = '\n'.join(TEMPLATE) if isinstance(TEMPLATE, list) else TEMPLATE
-    source = Template(template)
+    source = Template(DOCKERFILE_TEMPLATE)
```
Comment (chekunkov) [inline shub_image/init.py]: I think you also want to strip it to remove all  heading and trailing `\n`

**Pull request merged**